diff --git a/frameworks/memstat.conf b/frameworks/memstat.conf
new file mode 100644
index 0000000..62796cd
--- /dev/null
+++ b/frameworks/memstat.conf
@@ -0,0 +1,14 @@
+/dev
+/bin
+/usr/bin
+/usr/local/bin
+/usr/X11R6/bin
+/lib
+/usr/lib
+/usr/local/lib
+/usr/X11R6/lib
+/sbin
+/usr/sbin
+/usr/local/sbin
+/lib/libc5-compat
+/usr/lib/libc5-compat
\ No newline at end of file
diff --git a/products/berlin_device.mk b/products/berlin_device.mk
index 7423963..ab0569d 100644
--- a/products/berlin_device.mk
+++ b/products/berlin_device.mk
@@ -203,6 +203,7 @@ PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml
 
 PRODUCT_COPY_FILES += \
+    vendor/marvell/generic/frameworks/memstat.conf:system/etc/memstat.conf \
     vendor/marvell/generic/frameworks/wifi/dhcpcd.conf:system/etc/wifi/dhcpcd.conf \
     vendor/marvell/generic/frameworks/wifi/mef.conf:system/etc/wifi/mef.conf \
     vendor/marvell/generic/frameworks/wifi/p2p_supplicant.conf:system/etc/wifi/p2p_supplicant.conf \
diff --git a/system/core/rootdir/init.berlin.rc b/system/core/rootdir/init.berlin.rc
index 5a01838..fd6170a 100644
--- a/system/core/rootdir/init.berlin.rc
+++ b/system/core/rootdir/init.berlin.rc
@@ -208,3 +208,6 @@ service install_recovery /system/bin/install_recovery
    class main
    oneshot
 
+service memstat /system/vendor/bin/memstat-arm -l
+   class main
+   group root
